lib/Sema/TypeChecker.cpp

```diff
@@ -209,17 +209,11 @@ Type TypeChecker::getObjectLiteralParameterType(ObjectLiteralExpr *expr,
 }
 
 ModuleDecl *TypeChecker::getStdlibModule(const DeclContext *dc) {
-  if (StdlibModule)
-    return StdlibModule;
-
-  StdlibModule = Context.getStdlibModule();
-
-  if (!StdlibModule) {
-    return dc->getParentModule();
+  if (auto *stdlib = Context.getStdlibModule()) {
+    return stdlib;
   }
 
-  assert(StdlibModule && "no main module found");
-  return StdlibModule;
+  return dc->getParentModule();
```

Comment:
Is this reachable? If not, you can remove this method entirely and update all callers to just call Context.getStdlibModule()

Reply:
It should come up if we're compiling stdlib itself.  But there are so few callers now it's something I'll look into.